shell/platform/glfw/key_event_handler.cc

```diff
@@ -5,21 +5,106 @@
 #include "flutter/shell/platform/glfw/key_event_handler.h"
 
 #include <iostream>
+#include <vector>
 
 #include "flutter/shell/platform/common/cpp/client_wrapper/include/flutter/json_message_codec.h"
 
 static constexpr char kChannelName[] = "flutter/keyevent";
 
 static constexpr char kKeyCodeKey[] = "keyCode";
 static constexpr char kKeyMapKey[] = "keymap";
+static constexpr char kScanCodeKey[] = "scanCode";
+static constexpr char kModifiersKey[] = "modifiers";
 static constexpr char kTypeKey[] = "type";
+static constexpr char kToolkitKey[] = "toolkit";
+static constexpr char kUnicodeScalarValuesProduced[] =
+    "unicodeScalarValuesProduced";
+
+static constexpr char kLinuxKeyMap[] = "linux";
+static constexpr char kGLFWKey[] = "glfw";
 
-static constexpr char kAndroidKeyMap[] = "android";
 static constexpr char kKeyUp[] = "keyup";
 static constexpr char kKeyDown[] = "keydown";
 
+// Masks used for UTF-8 to UTF-32 conversion.
+static constexpr int kTwoByteMask = 0xC0;
+static constexpr int kThreeByteMask = 0xE0;
+static constexpr int kFourByteMask = 0xF0;
+
 namespace flutter {
 
+// Information about the UTF-8 encoded code point.
```

Comment:
Wrap this and the helper function in an anonymous namespace so they have internal linkage.

Reply:
Done